backend/LexBoxApi/GraphQL/CustomTypes/ProjectMembersVisibilityMiddleware.cs

```diff
@@ -0,0 +1,24 @@
+using HotChocolate.Resolvers;
+using LexCore.Entities;
+using LexCore.ServiceInterfaces;
+
+namespace LexBoxApi.GraphQL.CustomTypes;
+
+public class ProjectMembersVisibilityMiddleware(FieldDelegate next)
+{
+    public async Task InvokeAsync(IMiddlewareContext context, IPermissionService permissionService)
+    {
+        await next(context);
+        if (context.Result is List<ProjectUsers>)
+        {
+            var contextProject = context.Parent<Project>();
+            var projId = contextProject?.Id;
+            // If we don't have a project ID to use, have to assume it's confidential
+            if (projId is null || !await permissionService.CanViewProjectMembers(projId.Value))
```

Comment:
I'd rather throw an error, imagine this. You write a query for a projects users, if you leave off the ID then you get back no users, otherwise you get back the actual list? that's pretty confusing.

Reply:
Done.